sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
+<!-- no toc -->
+* [Authenticate with DefaultAzureCredential](#authenticate-with-defaultazurecredential "Authenticate with DefaultAzureCredential")
+<!-- TODO: Un-Comment if/when the sections are added>
+* [Define a custom authentication flow with ChainedTokenCredential](#define-a-custom-authentication-flow-with-chainedtokencredential "Define a custom authentication flow with ChainedTokenCredential")
+* [Async credentials](#async-credentials "Async credentials")
+</!-->
+
+### Authenticate with `DefaultAzureCredential`
+
+More details on configuring your environment to use `DefaultAzureCredential` can be found in the class's [reference documentation][default_cred_ref].
+
+This example demonstrates authenticating the `CosmosClient` from the [azure_data_cosmos][azure_data_cosmos] library using `DefaultAzureCredential`.
 
 ```rust
-use azure_core::credentials::TokenCredential;
+use azure_data_cosmos::CosmosClient;
 use azure_identity::DefaultAzureCredential;
-# use azure_core::{ClientOptions, Result};
-# use std::sync::Arc;
-# struct StorageAccountClient;
-# impl StorageAccountClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(StorageAccountClient))
-#     }
-# }
-# struct SecretClient;
-# impl SecretClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(SecretClient))
-#     }
-# }
 
 let credential = DefaultAzureCredential::new().unwrap();
-let storage_client = StorageAccountClient::new(
-    "https://myaccount.blob.storage.azure.net",
-    credential.clone(),
-    None,
-);
-let secret_client = SecretClient::new("https://myvault.keyvault.azure.net",
-    credential.clone(),
-    None,
-);
+let client = CosmosClient::new("https://myaccount.documents.azure.com/", credential, None).unwrap();
+```
+
+<!-- TODO: update this section on enabling interactive auth if/when it's configurable in the Rust library>
+#### Enable interactive authentication with `DefaultAzureCredential`
+
+By default, interactive authentication is disabled in `DefaultAzureCredential` and can be enabled with a keyword argument:
+
+```python
+DefaultAzureCredential(exclude_interactive_browser_credential=False)
+```
+
+When enabled, `DefaultAzureCredential` falls back to interactively authenticating via the system's default web browser when no other credential is available.
+</!-->
+
+<!-- TODO: update this section on specifying a user-assigned managed identity if/when it's configurable in the Rust library>
+#### Specify a user-assigned managed identity with `DefaultAzureCredential`
+
+Many Azure hosts allow the assignment of a user-assigned managed identity. To configure `DefaultAzureCredential` to authenticate a user-assigned managed identity, use the `managed_identity_client_id` keyword argument:
+
+```python
+DefaultAzureCredential(managed_identity_client_id=client_id)
```

Comment:
I think that should be `DefaultAzureCredential::new()` or `DefaultAzureCredential::new().unwrap()`